wagtail/images/views/images.py

```diff
@@ -7,45 +7,56 @@
 from django.template.response import TemplateResponse
 from django.urls import reverse
 from django.urls.exceptions import NoReverseMatch
+from django.utils.functional import cached_property
 from django.utils.http import urlencode
 from django.utils.translation import gettext as _
 from django.utils.translation import gettext_lazy, ngettext
 from django.views import View
 
 from wagtail.admin import messages
 from wagtail.admin.auth import PermissionPolicyChecker
-from wagtail.admin.models import popular_tags_for_model
+from wagtail.admin.filters import BaseMediaFilterSet
 from wagtail.admin.utils import get_valid_next_url_from_request, set_query_params
 from wagtail.admin.views import generic
+from wagtail.admin.widgets.button import HeaderButton
 from wagtail.images import get_image_model
 from wagtail.images.exceptions import InvalidFilterSpecError
 from wagtail.images.forms import URLGeneratorForm, get_image_form
 from wagtail.images.models import Filter, SourceImageIOError
 from wagtail.images.permissions import permission_policy
 from wagtail.images.utils import generate_signature
-from wagtail.models import Collection, Site
+from wagtail.models import Site
 
 permission_checker = PermissionPolicyChecker(permission_policy)
 
-INDEX_PAGE_SIZE = getattr(settings, "WAGTAILIMAGES_INDEX_PAGE_SIZE", 30)
+Image = get_image_model()
+
 USAGE_PAGE_SIZE = getattr(settings, "WAGTAILIMAGES_USAGE_PAGE_SIZE", 20)
 
 
+class ImagesFilterSet(BaseMediaFilterSet):
+    permission_policy = permission_policy
+
+    class Meta:
+        model = Image
+        fields = []
+
+
 class IndexView(generic.IndexView):
-    ENTRIES_PER_PAGE_CHOICES = sorted({10, 30, 60, 100, 250, INDEX_PAGE_SIZE})
     ORDERING_OPTIONS = {
-        "-created_at": _("Newest"),
-        "created_at": _("Oldest"),
-        "title": _("Title: (A -> Z)"),
-        "-title": _("Title: (Z -> A)"),
-        "file_size": _("File size: (low to high)"),
-        "-file_size": _("File size: (high to low)"),
+        "-created_at": gettext_lazy("Newest"),
+        "created_at": gettext_lazy("Oldest"),
+        "title": gettext_lazy("Title: (A -> Z)"),
+        "-title": gettext_lazy("Title: (Z -> A)"),
+        "file_size": gettext_lazy("File size: (low to high)"),
+        "-file_size": gettext_lazy("File size: (high to low)"),
     }
     default_ordering = "-created_at"
```

Comment:
On initial load of the view (where there is no `ordering` parameter in the URL, so `is_explicitly_ordered = False`), if you do a search, this will not be used (it will be ordered by relevance instead). However, as soon as you change the ordering (thus adding `ordering` parameter to the URL), the search will no longer be ordered by relevance. This is the current behaviour and I'm not sure the best way to go around this.

We could introduce an unset option for the ordering in the UI, but what label should it use? Showing "------" like Django usually does for unset <select> inputs doesn't look very good and is unclear. Showing something like "Default" also isn't very clear what the ordering will be. Meanwhile, showing something like "Relevance" is irrelevant (forgive the pun) when you're not doing a search.

This is technically also an issue in the table-based listings, but is less prominent as the ordering is indicated by the column headers. On such views, I think we're also missing the functionality of unsetting the ordering once you've chosen an explicit ordering. This means you won't be able to order by relevance if you change the ordering after searching.

For this reason, I decided to leave this as-is for now.